classes/test.php

```diff
@@ -282,14 +284,35 @@ public function getAdapter()
 
 	public function setPhpMocker(php\mocker $phpMocker = null)
 	{
-		$this->phpMocker = $phpMocker ?: new php\mocker();
+		$phpMocker = $phpMocker ?: new php\mocker();
```

Comment:
I think the polymorphism, as suggested by @mageekguy, is the best solution. It allows us to gracefully handle both mocker.